ui/app/modifiers/keyboard-shortcut.js

```diff
@@ -1,44 +1,31 @@
 import { inject as service } from '@ember/service';
 import Modifier from 'ember-modifier';
 import { registerDestructor } from '@ember/destroyable';
-import { assert } from '@ember/debug';
 
 export default class KeyboardShortcutModifier extends Modifier {
   @service keyboard;
+  @service router;
```

Comment:
`suggestion` `question`:  I'm noticing we're using the `router` service alongside `registerDestructor`. But, I'm wondering if we need all that code because its going to fire specifically after page load.

`element-modifiers` have their own lifecycle that ties to the lifecycle of a DOM element.

Take this example code:

```
export default modifier((element, [callback]) => {
  function handleClick(event) {
    if (!element.contains(event.target)) {
      callback();
    }
  }

  document.addEventListener("click", handleClick);

  return () => {
    document.removeEventListener("click", handleClick);
  };
});
```

The function tears down once the element is removed from the DOM (e.g. changing conditional variable or transitioning to a new route. I think the Keyboard Service could delete some code and be simplified if we follow this pattern.

But, I might be missing something, is there a reason why we're not using the element's lifecycle?

Reply:
(I would have loved to not have to do this; it was in fact my first attempt. But destroyables in Ember have timings that don't align with practical use for this at all)